src/mc_control/generic_gripper.cpp

```diff
@@ -279,4 +299,46 @@ double Gripper::opening() const
   return std::accumulate(percentOpen.begin(), percentOpen.end(), 0.0) / static_cast<double>(percentOpen.size());
 }
 
+bool Gripper::complete() const
+{
+  bool res = (targetQ == nullptr);
+  if(!res)
+  {
+    res = true;
+    for(size_t i = 0; i < actualQ.size(); ++i)
+    {
+      res = res && overCommandLimit[i];
+    }
+  }
+  return res;
+}
```

Comment:
I don't get what the loop is checking. It will return true if there is a motion in course (i.e. `targetQ != nullptr` and *all* active joints are over their command limits. But in such condition, the motion is actually set to reverse that situation.

It seems to me that returning `targetQ == nullptr` is enough.

Reply:
Ping @Dwaaap: Do you remember why you wrote the condition that way?